test/e2e/healthcheck_test.go

```diff
@@ -0,0 +1,196 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/test/e2e/helpers"
+)
+
+// Test Cases covering multiple creation and deletion of health checks
+var _ = Describe("Health Check Test", func() {
```

Comment:
```suggestion
var _ = Describe("Health Check Test", Serial, func() {
```

If intend to use ENV vars we can never run this in parallel see https://onsi.github.io/ginkgo/#mental-model-spec-decorators